cmd/util/ledger/util/util.go

```diff
@@ -89,3 +91,41 @@ func (a *AccountsAtreeLedger) AllocateStorageIndex(owner []byte) (atree.StorageI
 	}
 	return v, nil
 }
+
+type PayloadSnapshot struct {
+	Payloads map[flow.RegisterID]*ledger.Payload
+}
+
+var _ snapshot.StorageSnapshot = (*PayloadSnapshot)(nil)
+
+func NewPayloadSnapshot(payloads []*ledger.Payload) (*PayloadSnapshot, error) {
+	l := &PayloadSnapshot{
+		Payloads: make(map[flow.RegisterID]*ledger.Payload, len(payloads)),
+	}
+	for _, payload := range payloads {
+		key, err := payload.Key()
+		if err != nil {
+			return nil, err
+		}
+		id, err := KeyToRegisterID(key)
+		if err != nil {
+			return nil, err
+		}
+		l.Payloads[id] = payload
+	}
+	return l, nil
+}
+
+func (p PayloadSnapshot) Get(id flow.RegisterID) (flow.RegisterValue, error) {
+	value := p.Payloads[id]
+	return value.Value(), nil
+}
+
+// NopMemoryGauge is a no-op implementation of the MemoryGauge interface
+type NopMemoryGauge struct{}
```

Comment:
I am not 100% sure, but I don't think it's necessary to define a new Nop interface here for a memory gauge that doesn't do anything; the Cadence code should be set up such that `nil` can be passed to anywhere that expects a `MemoryGauge` if you don't care about tracking memory usage.